### PR TITLE
Add test that features are enabled

### PR DESCRIPTION
**Pull request by nealrichardson**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Assuming this test is being run (I couldn't find where in the build logs), it should fail, given the reports in #56 and #62.

### CODE REVIEW

**Comment by nealrichardson** [comment]
Assuming this fails the builds as expected, we could try applying https://github.com/nealrichardson/arrow/commit/e925a3feed332535cc36639c7368467423f3eaa0 as a patch and see if that fixes things. I wasn't able to confirm this in apache/arrow CI because it turns out that the conda packages we build nightly already have the features enabled correctly (compare https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=685963&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=b2a8456a-fb11-5506-ca32-5ccd32538dc0&l=394 here to https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=47681&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=d9b15392-e4ce-5e4c-0c8c-b69645229181&l=21736 on apache/arrow, where the feature flags are set correctly).

**Comment by h-vetinari** [comment]
Do we need to do something like in the upstream R [Makefile](https://github.com/apache/arrow/blob/apache-arrow-11.0.0/r/Makefile#L37-L38) to discover the deps correctly?

**Comment by h-vetinari** [comment]
> if `ArrowOptions.cmake`, which is generated in the `arrow-cpp` build, isn't found, we won't be able to determine which features are enabled in the C++ library, so we'll essentially assume they aren't present and won't turn any of them on in the R package.

OK cool. That file is [definitely](https://github.com/regro/libcfgraph/blob/master/artifacts/libarrow/conda-forge/linux-64/libarrow-11.0.0-h93537a5_14_cpu.json#L773) being packaged with `libarrow`, so it should just be a question of pointing the R build to `$PREFIX/lib/cmake/Arrow/ArrowOptions.cmake`?

**Comment by nealrichardson** [comment]
Right. And in theory the patch I referenced would do that.

**Comment by h-vetinari** [comment]
> Right. And in theory the patch I referenced would do that.

I was hoping it might be as easy as setting that variable correctly, but alas no. Shouldn't an environment variable be picked up by the R configure script though?

**Comment by nealrichardson** [comment]
Hmm, based on the build log output, `LIB_DIR` is being set in the script and overwriting the env var you set, but interestingly there's no `"` around that line like the lines above it: https://github.com/apache/arrow/blob/main/r/configure#L113

I don't understand the subtleties of `$PREFIX` here, but I just see that the `PKG_LIBS="${PKGCONFIG_LIBS}"` seems to do what is expected but `LIB_DIR=${FOUND_LIB_DIR}` does not, and we know from the log message printed by L109 that `FOUND_LIB_DIR` is literally `$PREFIX/lib` https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=695884&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=294.

File: recipe/build.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 set -ex
 
+export LIB_DIR=$PREFIX
```
Comment (h-vetinari) [inline recipe/build.sh:4]:
```suggestion
export LIB_DIR=$PREFIX/lib
```

**Comment by h-vetinari** [comment]
Turns out I had forgotten to add `/lib` to the `$PREFIX`, now it works! :)

I think I'm going to merge this to unblock the two issues that were reported; it should be a net improvement in all respects already, and we can still keep iterating in a new PR if necessary.

**Comment by nealrichardson** [comment]
> Hmm, based on the build log output, `LIB_DIR` is being set in the script and overwriting the env var you set, but interestingly there's no `"` around that line like the lines above it: https://github.com/apache/arrow/blob/main/r/configure#L113
> 
> I don't understand the subtleties of `$PREFIX` here, but I just see that the `PKG_LIBS="${PKGCONFIG_LIBS}"` seems to do what is expected but `LIB_DIR=${FOUND_LIB_DIR}` does not, and we know from the log message printed by L109 that `FOUND_LIB_DIR` is literally `$PREFIX/lib` https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=695884&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=294.

For posterity, it turns out that I was looking at the latest `main` branch, where this issue had been fixed: https://github.com/apache/arrow/pull/34229

So the workaround added here can be removed in the 12.0.0 release (likely happening within days). Having it here won't harm anything as it will get overwritten in the `configure` script, as I described. And the test we added here will confirm that it is safe to delete the workaround.

**Pull request merged**